rapidwright

```diff
@@ -0,0 +1,51 @@
+#!/bin/sh
+
+#  Copyright (c) 2023, Advanced Micro Devices, Inc.
+#  All rights reserved.
+# 
+#  Author: Eddie Hung, Advanced Micro Devices, Inc.
+# 
+#  This file is part of RapidWright.
+# 
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+# 
+#      http://www.apache.org/licenses/LICENSE-2.0
+# 
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+# Check that the main jar has been built
+# NOTE: Does not check that it is up-to-date
+MAIN_JAR=${BASH_SOURCE%/*}/build/libs/main.jar
+if [[ ! -f ${MAIN_JAR} ]]; then
+    echo "RapidWright not yet compiled. Please run './gradlew compileJava' from '`readlink -f ${BASH_SOURCE%/*}`'"
+    exit 1
+fi
+
+
+if [[ "$1" == "--help" || "$1" == "-h" ]]; then
+    echo "./rapidwright com.xilinx.rapidwright.* -- to execute main entrypoint of class"
+    echo "./rapidwright <application>            -- to execute a specific application"
+    echo "                                          (leave blank to see list)"
```

Comment:
Although I feel it is important that we provide a list of useful capabilities in RapidWright out-of-the-box to run, I'm not sure if presenting the user with a long list tasks is the right experience when running `./rapidwright`.  

Personally, I use the Jython interpreter often to test and prototype ideas and it would be convenient for me for `./rapidwright` without any options to default to an interactive Jython prompt with RapidWright packages pre-loaded.  The Jython interpreter also has tab-complete which allows one to explore the available classes and APIs. 
 
However, I can recognize presenting the Jython interpreter right off the bat could be foreign to some first time users, so I am open to suggestions.  What should be the initial experience?  Could there be a brief instructional message up front followed by a RapidWright-loaded Jython prompt?  Or, should we simply present a help message as most other command-line applications?

Reply:
> Although I feel it is important that we provide a list of useful capabilities in RapidWright out-of-the-box to run, I'm not sure if presenting the user with a long list of tasks is the right experience when running `./rapidwright`.

This wrapper script is currently modelled on the `./gradlew run` experience which prints out a list of all applications registered with `MainEntrypoint`.

> Personally, I use the Jython interpreter often to test and prototype ideas and it would be convenient for me for `./rapidwright` without any options to default to an interactive Jython prompt with RapidWright packages pre-loaded. The Jython interpreter also has tab-complete which allows one to explore the available classes and APIs.

I would say that running an all-in-one application (e.g. RWRoute) is likely to be what a beginner would want to do (as instructed by our tutorials) rather than dive into a Jython prompt.

It's also ambiguous as to what kind of prompt/shell would be expected from just `./rapidwright` -- is it JShell, Jython, or RapidWright's own language -- which is why I've qualified it with `./rapidwright Jython`.

> However, I can recognize presenting the Jython interpreter right off the bat could be foreign to some first time users, so I am open to suggestions. What should be the initial experience? Could there be a brief instructional message up front followed by a RapidWright-loaded Jython prompt? Or, should we simply present a help message as most other command-line applications?

I think a brief help message (maybe the one I have under `--help`) would make sense here: if you want to run an application, specify it. If you want to list all applications, use this option. If you want a Jython shell, use this other option.

Reply:
Roger that: 4d2b3e407fa8f993fa6ecd53394c8899a108713b